Attempt to invoke virtual method 'int android.view.WindowInsets.getSystemWindowInsetTop()' on a null object reference

### Description

Hi @jpudysz

It has happened only on Galaxy S8 (SM-G950U) using Android 9:

```
java.lang.RuntimeException: Unable to resume activity {com.APP_NAME/com.APP_NAME.MainActivity}: java.lang.reflect.InvocationTargetException
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4016)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4048)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1950)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7073)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:294)
        at expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:168)
        at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1412)
        at android.app.Activity.performResume(Activity.java:7558)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4008)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4048)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1950)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7073)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int android.view.WindowInsets.getSystemWindowInsetTop()' on a null object reference
        at com.unistyles.UnistylesInsets.getBaseInsets(Insets.kt:53)
        at com.unistyles.UnistylesInsets.getCurrentInsets(Insets.kt:21)
        at com.unistyles.UnistylesInsets.get(Insets.kt:17)
        at com.unistyles.UnistylesConfig.getAppLayoutConfig(Config.kt:65)
        at com.unistyles.UnistylesConfig.hasNewLayoutConfig(Config.kt:30)
        at com.unistyles.Platform.hasNewLayoutConfig(Platform.kt:9)
        at com.unistyles.UnistylesModule.onLayoutConfigChange(UnistylesModule.kt:100)
        at com.unistyles.UnistylesModule.onHostResume(UnistylesModule.kt:243)
        at com.facebook.react.bridge.ReactContext.onHostResume(ReactContext.java:308)
        at com.facebook.react.ReactInstanceManager.moveToResumedLifecycleState(ReactInstanceManager.java:765)
        at com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:663)
        at com.facebook.react.ReactInstanceManager.onHostResume(ReactInstanceManager.java:613)
        at com.facebook.react.ReactDelegate.onHostResume(ReactDelegate.java:93)
        at com.facebook.react.ReactActivityDelegate.onResume(ReactActivityDelegate.java:132)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at expo.modules.ReactActivityDelegateWrapper.invokeDelegateMethod(ReactActivityDelegateWrapper.kt:294)
        at expo.modules.ReactActivityDelegateWrapper.onResume(ReactActivityDelegateWrapper.kt:168)
        at com.facebook.react.ReactActivity.onResume(ReactActivity.java:58)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1412)
        at android.app.Activity.performResume(Activity.java:7558)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4008)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4048)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1950)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7073)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
```

### Steps to reproduce

1. This is happening on app/device resume, according to Bugsnag breadcrumbs the device was in sleep mode and when it waked in the app it crashed.

### Snack or a link to a repository (optional)

_No response_

### Unistyles version

2.5.3

### React Native version

0.73.6

### Platforms

Android

### Engine

Hermes

### Architecture

Paper (old)

--- a/android/src/main/java/com/unistyles/Insets.kt
+++ b/android/src/main/java/com/unistyles/Insets.kt
@@ -48,7 +48,7 @@ class UnistylesInsets(private val reactApplicationContext: ReactApplicationConte
 
         // available from Android 6.0
         val window = reactApplicationContext.currentActivity?.window ?: return Insets(0, 0, 0, 0)
-        val systemInsets = window.decorView.rootWindowInsets
+        val systemInsets = window.decorView.rootWindowInsets ?: return Insets(0, 0, 0, 0)
 
         val top = (systemInsets.systemWindowInsetTop / density).roundToInt()
         val bottom = (systemInsets.systemWindowInsetBottom / density).roundToInt()

--- a/android/src/main/java/com/unistyles/UnistylesModule.kt
+++ b/android/src/main/java/com/unistyles/UnistylesModule.kt
@@ -67,13 +67,14 @@ class UnistylesModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
 
     @Deprecated("Deprecated in Java")
     override fun onCatalystInstanceDestroy() {
-        val activity = currentActivity ?: return
-
         this.stopLayoutListener()
         reactApplicationContext.unregisterReceiver(configurationChangeReceiver)
         runnable?.let { drawHandler.removeCallbacks(it) }
         reactApplicationContext.removeLifecycleEventListener(this)
-        this.nativeDestroy()
+
+        if (this.isCxxReady) {
+            this.nativeDestroy()
+        }
     }
 
     //endregion
@@ -144,7 +145,9 @@ class UnistylesModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
 
             false
         } catch (e: Exception) {
-            false
+            this.isCxxReady = false
+
+            return false
         }
     }
 
@@ -240,7 +243,6 @@ class UnistylesModule(reactContext: ReactApplicationContext) : ReactContextBaseJ
     override fun onHostResume() {
         if (isCxxReady) {
             this.onConfigChange()
-            this.onLayoutConfigChange()
         }
 
         this.setupLayoutListener()

